uroot/root.go

```diff
@@ -63,40 +67,45 @@ var (
 		"CGO_ENABLED":     "0",
 	}
 
-	dirs = []dir{
-		{name: "/proc", mode: os.FileMode(0555)},
-		{name: "/sys", mode: os.FileMode(0555)},
-		{name: "/buildbin", mode: os.FileMode(0777)},
-		{name: "/ubin", mode: os.FileMode(0777)},
-		{name: "/tmp", mode: os.FileMode(0777)},
-		{name: "/env", mode: os.FileMode(0777)},
-		{name: "/etc", mode: os.FileMode(0777)},
-		{name: "/tcz", mode: os.FileMode(0777)},
-		{name: "/dev", mode: os.FileMode(0777)},
-		{name: "/lib", mode: os.FileMode(0777)},
-		{name: "/usr/lib", mode: os.FileMode(0777)},
-		{name: "/go/pkg/linux_amd64", mode: os.FileMode(0777)},
-		// This is for uroot packages. Is this a good idea? I don't know.
-		{name: "/pkg", mode: os.FileMode(0777)},
-	}
-	devs = []dev{
+	namespace = []interface{}{
+		dir{name: "/proc", mode: os.FileMode(0555)},
```

Comment:
Instead of doing this, I suggest this:

type Creator interface {
  Create() error
  fmt.Stringer // for debugging log.Printf below
}

Make dir, mount, symlink, etc, implement Create.

e.g.

```go
type dir struct {
  name string
  mode uint32
}

func (d dir) Create() error {
  // Code from .(dir) case below:
  return os.MkdirAll(d.name, d.mode)
}
```

Then, this array should become an array of Creators, and you can simply 

```go
for _, c := range creators {
  if err := c.Create(); err != nil {
    log.Printf("Error creating %s: %v", c, err)
  }
}
```